client/js/lounge.js

```diff
@@ -1045,8 +1041,12 @@ $(function() {
 	});
 
 	sidebar.on("click", "#sign-out", function() {
+		socket.emit("sign-out", storage.get("token"));
 		storage.remove("token");
-		location.reload();
+
+		if (!socket.connected) {
```

Comment:
Hmmm, is this because we get a notification from the server that we have logged out and then we refresh if we already are connected? Also, the emit for sign-out won't work if we aren't connected.